app/uk/gov/hmrc/agentclientrelationships/auth/AuthActions.scala

```diff
@@ -34,27 +35,29 @@ trait AuthActions extends AuthorisedFunctions {
 
   override def authConnector: AuthConnector
 
-  private def getEnrolmentInfo(enrolment: Set[Enrolment], enrolmentKey: String, identifier: String): Option[String] =
-    enrolment.find(_.key equals enrolmentKey).flatMap(_.identifiers.find(_.key equals identifier).map(_.value))
+  protected type AsyncPlayUserRequest = Request[AnyContent] => Future[Result]
 
-  protected type AsyncPlayUserRequest = Request[AnyContent] => AgentOrClientRequest[AnyContent] => Future[Result]
-
-  def AuthorisedAgent[A](body: AsyncPlayUserRequest): Action[AnyContent] = Action.async {
+  def AuthorisedAgentOrClient(arn: Arn, clientId: TaxIdentifier)(body: AsyncPlayUserRequest): Action[AnyContent] = Action.async {
     implicit request =>
       implicit val hc = HeaderCarrierConverter.fromHeadersAndSession(request.headers, None)
       authorised(AuthProviders(GovernmentGateway)).retrieve(allEnrolments and affinityGroup) {
-        case enrol ~ affinityG =>
-          val maybeIdentifier: Option[TaxIdentifier] = affinityG match {
-            case Some(AffinityGroup.Agent) => getEnrolmentInfo(enrol.enrolments, "HMRC-AS-AGENT", "AgentReferenceNumber").map(arn => Arn(arn))
-            case _ => getEnrolmentInfo(enrol.enrolments, "HMRC-MTD-IT", "MTDITID").map(mtdItId => MtdItId(mtdItId))
+        case enrol ~ affinityG => {
+
+          val requiredIdentifier = affinityG match {
+            case Some(AffinityGroup.Agent) => arn
```

Comment:
`getEnrolmentInfo` method returns an arn from the enrolments. SHould we need to check against the user supplied arn?

Reply:
we could possibly rewrite it using `Enrolment.withIdentifier` predicates joined with OR